### PR TITLE
Fix: User management doesn't work

### LINKED ISSUE #14640: [Bug]: User management doesn't work
### Expected behavior

Users can be saved, created and deleted 

### Actual behavior

```
Status: 500 | Internal Server Error
URL: /admin/user/delete
Method: DELETE
Message: Cannot instantiate abstract class Pimcore\Model\User\AbstractUser
Trace: 
in /var/www/html/dev/pimcore/pimcore/models/User/AbstractUser.php:48
#0 /var/www/html/dev/pimcore/pimcore/bundles/AdminBundle/src/Controller/Admin/UserController.php(246): Pimcore\Model\User\AbstractUser::getById(5)
#1 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(163): Pimcore\Bundle\AdminBundle\Controller\Admin\UserController->deleteAction(Object(Symfony\Component\HttpFoundation\Request))
#2 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(74): Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object(Symfony\Component\HttpFoundation\Request), 1)
#3 /var/www/html/vendor/symfony/http-kernel/Kernel.php(184): Symfony\Component\HttpKernel\HttpKernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#4 /var/www/html/public/index.php(36): Symfony\Component\HttpKernel\Kernel->handle(Object(Symfony\Component\HttpFoundation\Request))
#5 {main}
```

### Steps to reproduce

Try to save, create or delete an user

### PR DESCRIPTION
**Pull request by blankse**
Fix: https://github.com/pimcore/pimcore/issues/14640
Followup to https://github.com/pimcore/pimcore/pull/14441

### CODE REVIEW

File: bundles/AdminBundle/src/Controller/Admin/UserController.php [hunk 1]
```diff
@@ -272,16 +272,12 @@ public function deleteAction(Request $request): JsonResponse
...
-     * @return JsonResponse
-     *
      * @throws \Exception
      */
     public function updateAction(Request $request): JsonResponse
     {
-        /** @var User|null $user */
-        $user = User\UserRole::getById((int)$request->get('id'));
+        /** @var User|User\Role|null $user */
+        $user = User\UserRole::getById($request->request->getInt('id'));
```
Comment (mattamon) [inline bundles/AdminBundle/src/Controller/Admin/UserController.php:280]: Are we sure this is always in the request? 
We had some problems with other changes that the data was not always in the request but sometimes in the query e.g.

File: models/User/AbstractUser.php [hunk 2]
```diff
@@ -45,7 +45,13 @@ public static function getById(int $id): static|null
             if (\Pimcore\Cache\RuntimeCache::isRegistered($cacheKey)) {
                 $user = \Pimcore\Cache\RuntimeCache::get($cacheKey);
             } else {
-                $user = new static();
+                $reflectionClass = new \ReflectionClass(static::class);
```
Comment (mattamon) [inline models/User/AbstractUser.php:48]: Do we really need reflection here?
It seems like more of a problem in the UserController in the delete function, where we call the AbstractUser::getById directly. Shouldn't we tackle the problem there?
https://github.com/pimcore/pimcore/blob/11.x/bundles/AdminBundle/src/Controller/Admin/UserController.php#L246

**Review by mattamon — CHANGES_REQUESTED** [review]

File: models/User/AbstractUser.php (hunk 2 above)
Comment (blankse) [reply models/User/AbstractUser.php:48]: I think it should still be possible to load role or user with a parent class (both classes are static now). The same we did in Document to load via PageSnippet abstract class:
https://github.com/pimcore/pimcore/blob/b23b847c42a79fce4ed6905570283a04ae2b221f/models/Document.php#L196-L201

File: bundles/AdminBundle/src/Controller/Admin/UserController.php (hunk 1 above)
Comment (blankse) [reply bundles/AdminBundle/src/Controller/Admin/UserController.php:280]: I looked in the JS files. The id was alwas set in POST data.

**Review by mattamon — APPROVED** [review]
LGTM, using the same approach as in Document for the PageSnippet

**Pull request merged**